tensilelite/Tensile/SolutionStructs/Solution.py

```diff
@@ -1701,11 +1701,18 @@ def calcLdsBlockSizePerPad(lrvw: int) -> int:
             LdsBlockSizePerPadB = 0
 
         # set LdsBlockSizePerPadA,B=0 for DirectToLds or DirectToVgpr
-        if state["DirectToLds"] or state["DirectToVgprA"]:
+        if state["DirectToVgprA"]:
           LdsBlockSizePerPadA = 0
-        if state["DirectToLds"] or state["DirectToVgprB"]:
+        if state["DirectToVgprB"]:
           LdsBlockSizePerPadB = 0
 
+        if state["DirectToLds"]:
+          bpeA = state["ProblemType"]["DataTypeA"].numBytes()
+          bpeB = state["ProblemType"]["DataTypeB"].numBytes()
+          LdsBlockSizePerPadA = (state[f"GlobalReadVectorWidthA"] * bpeA) * state["WavefrontSize"]
```

Comment:
Don't we need to consider UnrollMajorLDSA/B here?

Reply:
Regarding your comment below
> Regarding the UnrollMajorLDS comments, I am not sure if we need to consider it. This is mainly due to DTL supporting very limited forms of padding. Does UnrollMajorLDS{X} mean that the layout for X is contiguous in the accumulation dimension?

Basically, LDS padding strategy is different for UnrollMajorLDS = False and True cases.
UnrollMajorLDS = False case, we basically do not need LDS padding because Each thread reads data horizontally and does not cause LDS bank conflicts.
(but, we have some exceptions for this logic. You can learn more details from LDS_pad auto calculation code)
So, it is better not to enable LDS pad and LdsBlockSizePerPadA in UnrollMajorLDS = False case.
(means set 0 to them)

Reply:
Thanks for the explanations, I've updated the `LdsBlockSizePerPadX` calculations to use the existing logic and round to the nearest instruction load size ( `bytes loaded per thread * wave size`) when DTL is enabled.

Reply:
I do not think it is correct.
LdsBlockSizePerPadX should be multiple of DTL load unit (for each wave).

Reply:
Oops, yes, I mistook lds padding and ldsblocksizeperpad. Reverted.